src/element.ts

```diff
@@ -218,10 +218,7 @@ export class Element {
 
   /** Apply the element style to `context`. */
   applyStyle(context: RenderContext | undefined = this.context, style: ElementStyle = this.getStyle()): this {
-    if (Object.keys(style).length == 0) return this;
     if (!context) return this;
-
-    context.save();
```

Comment:
I'm OK with this new approach. The only risk is if someone manually called applyStyle(...) in their v4 compatible code, they'll expect that context.save() has been called. But with this new approach, it won't be called.

But we're removing restoreStyle() below, so anyone who manually calls applyStyle() / restoreStyle() will notice very quickly that their code is broken since it won't compile.

Reply:
Agree